linux/vhba_hot_add_remove/freebsd_create_disk_partition.yml

```diff
@@ -78,6 +67,103 @@
+        test_disk_partition_uuid: "{{ partition_uuid }}"
+
+    - name: "Get guest disk list before reboot"
+      include_tasks: ../utils/get_device_list.yml
+      vars:
+        guest_device_type: "disk"
+
+    - name: "Set the fact of guest disk list before reboot"
+      ansible.builtin.set_fact:
+        guest_disk_list_before_reboot: "{{ guest_device_list }}"
+
+    - name: "Reboot guest OS to check file content again"
+      include_tasks: ../utils/reboot.yml
+
+    - name: "Get guest disk list after reboot"
+      include_tasks: ../utils/get_device_list.yml
+      vars:
+        guest_device_type: "disk"
+
+    - name: "Set the fact of guest disk list after reboot"
+      ansible.builtin.set_fact:
+        guest_disk_list_after_reboot: "{{ guest_device_list }}"
+
+    - name: "Set fact of changed disk list after reboot"
+      ansible.builtin.set_fact:
+        guest_disk_changed_list: "{{ guest_disk_list_before_reboot | difference(guest_disk_list_after_reboot) }}"
+
+    - name: "Set the fact of test disk if it's changed after reboot"
+      when: guest_disk_changed_list | length > 0
+      block:
+        - name: "Get disk partition info by UUID on FreeBSD"
+          include_tasks: ../utils/freebsd_get_partition_info.yml
+          vars:
+            disk_partition_uuid: "{{ test_disk_partition_uuid }}"
+
+        - name: "Check partition with UUID {{ new_partition_uuid }} existing on FreeBSD"
+          ansible.builtin.assert:
+            that:
+              - disk_partition_info.name is defined
+              - disk_partition_info.name
+              - disk_partition_info.config.rawuuid is defined
+              - disk_partition_info.config.rawuuid == test_disk_partition_uuid
+            fail_msg: "Failed to get partition by UUID {{ test_disk_partition_uuid }} on FreeBSD"
+
+        - name: "Print test disk partition info in guest OS after it'changed"
+          ansible.builtin.debug: var=disk_partition_info
+
+        - name: "Change name of test disk after it's changed"
+          ansible.builtin.set_fact:
+            new_guest_disk_info: > 
+              {{ new_guest_disk_info | combine({
+                    'name': disk_partition_info.name[0:-2]
+                }) 
+              }}
+        
+        - name: "Change partition name and path of test disk after it's changed"
+          ansible.builtin.set_fact:
+            partition_name: "{{ disk_partition_info.name }}"
+            partition_device_path: "/dev/{{ disk_partition_info.name }}"
+
+        - name: "Print test disk info in guest OS after it'changed"
+          ansible.builtin.debug: var=new_guest_disk_info
+
+- name: "Create and check {{ partition_fstype }} filesystem on partition {{ partition_device_path }}"
+  ansible.builtin.shell: "newfs -EU {{ partition_device_path }} && sleep 5 && fstyp {{ partition_device_path }}"
+  delegate_to: "{{ vm_guest_ip }}"
+  ignore_errors: true
+  register: partition_result
+  retries: 5
+  delay: 2
+  until:
```

Comment:
It is better to add loop into the shell command instead of retries to this `ansible.builtin.shell` task. And `partition_result` as the var name is not proper here. This task is to create filesystem on the partition not create partition. Maybe you can change the registered var to `freebsd_newfs_result`.